### DISCUSSION TITLE
Can Copilot work with private Bitbucket repositories?

### DISCUSSION THREAD

**Discussion by vadi2** [community/community#72275] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Can Copilot work with private Bitbucket repositories?

**Comment by Akash1134** [comment, marked as answer]
Hi @vadi2 ,

Copilot is designed to work with GitHub repositories and does not currently support private Bitbucket repositories. However, Copilot is constantly evolving and there have been cases where users have experimented with it. Here's [one such article](https://medium.com/@chrislondon/supercharge-your-coding-with-githubs-copilot-even-if-you-use-bitbucket-1a39554f4a8d#:~:text=While%20Copilot%20is%20primarily%20associated,for%20a%20seamless%20coding%20experience.) I found that might be helpful. 

Thanks!

**Reply by ivarec** [reply]
But the performance is likely degraded, no? I imagine that GitHub's integration will have interesting ways to do RAG and learn about the whole code base, and not just operate locally as I suppose the "pure Copilot" + VSCode extension will do.

**Reply by vadi2** [reply]
From what I know, they do not. It is Sourcegraph Cody that does RAG - and it really does help.

**Reply by davecheney** [reply]
Code completion does not require your source code to be located in a particular VCS or source code hosting service. Copilot will work just as well on a file that has not been committed. It'll work just as well on a file that hasn't even been saved.

**Comment by sanjay-ar** [comment]
Copilot excels in supporting GitHub repositories but doesn't natively integrate with private Bitbucket repositories. However, given Copilot's adaptability, there have been instances where users have explored workarounds. Check out this article I stumbled upon; it might offer some insights.

You're welcome!
